e2e/app/page/authenticated-page.ts

```diff
@@ -21,27 +20,33 @@ export default abstract class AuthenticatedPage extends BasePage {
   }
 
   protected async isSignedIn(): Promise<boolean> {
-    await this.page.waitForSelector(SELECTOR.signedInIndicator);
-    await this.page.waitForSelector(SELECTOR.logo, {visible: true});
-    return true;
+    try {
+      await this._page.waitForSelector(SELECTOR.signedInIndicator);
+      await this._page.waitForSelector(SELECTOR.logo, {visible: true});
+      return true;
+    } catch (err) {
+      return false;
+    }
   }
 
   /**
    * Method to be implemented by children classes.
    * Check whether current page has specified web elements.
    */
-  abstract async isLoaded(): Promise<boolean>
+  async isLoaded(): Promise<boolean> {
+    throw new Error('Implement isLoaded().');
+  }
 
   /**
    * Wait until current page is loaded and without spinners spinning.
    */
   async waitForLoad(): Promise<this> {
     if (!await this.isLoaded()) {
-      await savePageToFile(this.page, 'PageIsNotLoaded');
-      await takeScreenshot(this.page, 'PageIsNotLoaded');
-      throw new Error('Page isLoaded() failed.');
+      await savePageToFile(this._page);
+      await takeScreenshot(this._page);
+      const title = await this._page.title();
+      throw new Error(`${title}: page not loaded correctly.`);
```

Comment:
Please give the path to the screenshot and html in the error message (on separate lines would help).

Reply:
`savePageToFile` and `takeScreenshot` functions already log that.
e.g.
```
console.log utils/save-file-utils.ts:23
    Saved file logs/html/CreateWorkspace_2020-05-18T21:58:24.461Z.html

  console.log utils/save-file-utils.ts:41
    Saved screenshot logs/screenshot/CreateWorkspace_2020-05-18T21:58:24.593Z.png
```